tflearn/utils.py

```diff
@@ -384,3 +384,52 @@ def autoformat_padding(padding):
         return str.upper(padding)
     else:
         raise Exception("Unknow padding! Accepted values: 'same', 'valid'.")
+
+
+# Auto format filter size
+# Output shape: (rows, cols, input_depth, out_depth)
+def autoformat_filter_conv3d(fsize, in_depth, out_depth):
+    if isinstance(fsize, int):
+        return [fsize, fsize, fsize, in_depth, out_depth]
+    elif isinstance(fsize, (tuple, list)):
+        if len(fsize) == 3:
+            return [fsize[0], fsize[1],fsize[2], in_depth, out_depth]
+        else:
+            raise Exception("filter length error: " + str(len(fsize))
+                            + ", only a length of 3 is supported.")
+    else:
+        raise Exception("filter format error: " + str(type(fsize)))
+
+
+# Auto format stride for 3d convolution
+def autoformat_stride_3d(strides):
+    if isinstance(strides, int):
+        return [1, strides, strides, 1]
```

Comment:
I am not doing that because that would mean depth = 1, which would be equivalent to conv2d . So I  thought a  cube of unit volume would better be convincing for a single valued stride.